include/nil/actor/zk/commitments/polynomial/kzg.hpp

```diff
@@ -30,101 +30,786 @@
 #include <boost/accumulators/accumulators.hpp>
 
 #include <nil/actor/math/polynomial/polynomial.hpp>
+#include <nil/actor/math/polynomial/lagrange_interpolation.hpp>
 #include <nil/crypto3/algebra/type_traits.hpp>
 #include <nil/crypto3/algebra/algorithms/pair.hpp>
-#include <nil/crypto3/algebra/pairing/pairing_policy.hpp>
-
-using namespace nil::crypto3::math;
+#include <nil/crypto3/algebra/multiexp/multiexp.hpp>
+#include <nil/crypto3/algebra/multiexp/policies.hpp>
+#include <nil/crypto3/algebra/curves/detail/marshalling.hpp>
+#include <nil/crypto3/algebra/marshalling.hpp>
 
+#include <nil/actor/zk/transcript/fiat_shamir.hpp>
 #include <nil/actor/math/polynomial/polynomial.hpp>
 
+#include <nil/actor/zk/commitments/batched_commitment.hpp>
+
+using namespace nil::actor::math;
+
+using namespace nil::actor;
+
 namespace nil {
     namespace actor {
         namespace zk {
             namespace commitments {
+                /**
+                 * @brief The KZG Polynomial Commitment with Fiat-Shamir heuristic.
+                 *
+                 * References:
+                 * "Constant-Size Commitments to Polynomials and
+                 * Their Applications",
+                 * Aniket Kate, Gregory M. Zaverucha, and Ian Goldberg,
+                 * <https://www.iacr.org/archive/asiacrypt2010/6477178/6477178.pdf>
+                 */
                 template<typename CurveType>
                 struct kzg {
 
                     typedef CurveType curve_type;
-                    typedef algebra::pairing::pairing_policy<curve_type> pairing;
                     typedef typename curve_type::gt_type::value_type gt_value_type;
 
-                    using base_field_value_type = typename curve_type::base_field_type::value_type;
-                    using commitment_key_type = std::vector<typename curve_type::template g1_type<>::value_type>;
+                    using multiexp_method = typename crypto3::algebra::policies::multiexp_method_BDLO12;
+                    using field_type = typename curve_type::scalar_field_type;
+                    using scalar_value_type = typename curve_type::scalar_field_type::value_type;
+                    using single_commitment_type = std::vector<typename curve_type::template g1_type<>::value_type>;
                     using verification_key_type = typename curve_type::template g2_type<>::value_type;
                     using commitment_type = typename curve_type::template g1_type<>::value_type;
                     using proof_type = commitment_type;
 
                     struct params_type {
-                        std::size_t a;
+                        using commitment_type = typename curve_type::template g1_type<>::value_type;
+                        using field_type = typename curve_type::scalar_field_type;
+
+                        single_commitment_type commitment_key;
+                        verification_key_type verification_key;
+                        params_type() {}
```

Comment:
Why we need such constructor? It's default as I understand. So mb `= default;`

Reply:
Fixed.